analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ImplementSerializationMethodsCorrectly.CSharp11.cs

```diff
@@ -11,4 +11,13 @@ internal class TestCases
     }
 
     public class GenericAttribute<T> : Attribute { }
+
+    interface IMyInterface
+    {
+        [OnSerializing]
+        static virtual void OnSerializingStaticVirtual(StreamingContext context) { } // Noncompliant {{Make this method non-public and non-static.}}
```

Comment:
If I am not mistaken an interface cannot have the [Serializable] attribute which means these methods will not be called anyway, so I would document these as FPs.